internal/header/feature_list_element.go

```diff
@@ -0,0 +1,232 @@
+/* Copyright 2019 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package header
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	RegexpPredicateBag = regexp.MustCompile(regexp.QuoteMeta(`^\[\s{0,1}(.*)\s{0,1}\](;(\+(\d{0,1}\.{0,1}\d{1,3}))?(\-(\d{0,1}\.{0,1}\d{1,3}))?)?$`))
+	RegexpPredicate    = regexp.MustCompile(regexp.QuoteMeta(`^(\!)?(\w+)((\=|\!\=)((\{(\d)?\-(\d)?\})|([0-9a-zA-Z\.]+)))?(;(\+(\d{0,1}\.{0,1}\d{1,3}))?(\-(\d{0,1}\.{0,1}\d{1,3}))?)?$`))
+)
+
+// FeatureListElement represents a single element within a feature list. An
+// element can either be a feature predicate or feature predicate bag.
+type FeatureListElement interface {
+	evaluator
+	fmt.Stringer
+
+	TrueImprovement() TrueImprovement
+	FalseDegradation() FalseDegradation
+}
+
+type featureListElement struct {
```

Comment:
add comment for `featureListElement`, as well as for other types and methods in this file.

Reply:
done!